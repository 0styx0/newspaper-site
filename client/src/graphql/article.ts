import gql from 'graphql-tag';

const ArticleQuery = gql`
    query ArticleQuery($issue: Int!, $url: String!) {
        articles(issue: $issue, url: $url) {
            id,
            article,
            canEdit,
            tags,
            comments {
                id
                content,
                dateCreated,
                canDelete,
                author {
                    fullName,
                    profileLink
                }
            }
        }
    }

`;

const ArticleCreate = gql`
    mutation ArticleCreate($tags: [String], $url: String!, $article: String!) {
        createArticle(tags: $tags, url: $url, article: $article) {
            issue
            url
        }
    }
`;

export {
    ArticleQuery,
    ArticleCreate
};
