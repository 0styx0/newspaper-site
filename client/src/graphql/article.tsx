import { gql } from 'react-apollo';

const ArticleQuery = gql`
    query ArticleQuery($issue: Int!, $url: String!) {
        articles(issue: $issue, url: $url) {
            id,
            article,
            canEdit,
            comments {
                id
                content,
                dateCreated,
                canEdit,
                author {
                    fullName,
                    profileLink
                }
            }
            tags {
                all
            }
        }
    }

`;

export {
    ArticleQuery
};
