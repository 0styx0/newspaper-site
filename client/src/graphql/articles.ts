import gql from 'graphql-tag';

const ArticleQuery = gql`

    query ArticleQuery($issue: ID) {
        issues(limit: 1, num: $issue) {
          max
          num
          articles {
            tags
            url
            id
            displayOrder
            dateCreated
            views
            author {
                fullName
                profileLink
            }
          }
        }
    }
`;

const ArticlePreviewIssueQuery = gql`

    query ArticleIssueQuery($issue: ID) {
        issues(limit: 1, num: $issue) {
          articles {
            url
            issue
            displayOrder
            views
            lede
            images(slide: true) {
                url
            }
          }
        }
    }
`;

const ArticlePreviewTagQuery = gql`

    query ArticleTagQuery($issue: Int, $tag: String) {
        articles(issue: $issue, tag: $tag) {
          url
          issue
          displayOrder
          views
          lede
          images(slide: true) {
              url
          }
        }
    }
`;

const ArticleUpdate = gql`
    mutation ArticleUpdate($data: [UpdateArticle], $password: String) {
        updateArticles(data: $data, password: $password) {
            id
            tags
            displayOrder
        }
    }
`;

const EditArticle = gql`
    mutation EditArticle($id: ID, $article: String) {
        editArticle(id: $id, article: $article) {
            id
            tags
            displayOrder
        }
    }
`;

const ArticleDelete = gql`

    mutation deleteArticles($ids: [ID], $password: String!) {
        deleteArticles(ids: $ids, password: $password) {
            id
        }
    }

`;

export {
    ArticleQuery,
    ArticlePreviewIssueQuery,
    ArticlePreviewTagQuery,
    ArticleUpdate,
    EditArticle,
    ArticleDelete
};