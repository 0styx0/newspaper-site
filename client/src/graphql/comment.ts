import { gql } from 'react-apollo';

const CommentCreate = gql`

    mutation CommentCreate($artId: ID!, $content: String!) {
        createComment(artId: $artId, content: $content) {
            id
        }
    }
`;

export {
    CommentCreate
};