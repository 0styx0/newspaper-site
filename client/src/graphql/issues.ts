import gql from 'graphql-tag';


const IssueQuery = gql`
    query IssueQuery {
        issues {
            num
            name
            views
            datePublished
            public
            canEdit
        }
    }
`;

const IssueUpdate = gql`
    mutation updateIssue($name: String, $public: Boolean, $password: String!) {
        updateIssue(name: $name, public: $public, password: $password) {
            name
            public
        }
    }
`;

const IssueInfoQuery = gql`
    query IssueInfo($num: ID) {
      issues(num: $num, limit: 1) {
          num
          max
          name
      }
    }
`;



export {
    IssueQuery,
    IssueInfoQuery,
    IssueUpdate
};
