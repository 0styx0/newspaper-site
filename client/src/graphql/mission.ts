import gql from 'graphql-tag';

const MissionEdit = gql`
    mutation editMission($mission: String!) {
        editMission(mission: $mission) {
            mission
        }
    }
`;

const MissionQuery = gql`
    query missionQuery {
        mission {
            mission
            canEdit
        }
    }
`;

export {
    MissionEdit,
    MissionQuery
};