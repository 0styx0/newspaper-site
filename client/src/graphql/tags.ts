import gql from 'graphql-tag';

const TagQuery = gql`

    query allTags {
        allTags
    }
`;

const TagCreate = gql`

    mutation createTag($tag: String!) {
        createTag(tag: $tag) {
            tag
        }
    }
`;

export {
    TagQuery,
    TagCreate
};
