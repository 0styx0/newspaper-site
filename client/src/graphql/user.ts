import { gql } from 'react-apollo';

const UserQuery = gql`

    query users($profileLink: String) {
        users(profileLink: $profileLink) {
            articles {
                id
                url
                dateCreated
                tags
                views
                issue
                canEdit
            }
            views
            level
            id
            fullName
            canEdit
        }
    }
`;

const PrivateUserQuery = gql`

    query users($profileLink: String) {
        users(profileLink: $profileLink) {
            notifications
            twoFactor
            email
            id
        }
    }
`;

const UserUpdate = gql`
    mutation updateProfile(
        $notifications: Boolean, $twoFactor: Boolean, $newPassword: String, $password: String!
    ) {
        updateProfile(
            notifications: $notifications, twoFactor: $twoFactor,
            newPassword: $newPassword, password: $password
        ) {
            id
        }
    }
`;

const RecoverPassword = gql`
    mutation recoverPassword($email: String!, $authCode: String!, $username: String!) {
        recoverPassword(email: $email, authCode: $authCode, username: $username) {
            message
        }
    }
`;

const UserLogin = gql`
    mutation login($username: String!, $password: String!) {
        login(username: $username, password: $password) {
            jwt
        }
    }
`;

// to be used only after user is logged in
const UserVerifyEmail = gql`
    mutation verifyEmail($authCode: String!) {
        verifyEmail(authCode: $authCode) {
            jwt
        }
    }
`;

const UserCreate = gql`
    mutation createUser(
        $username: String!,
        $email: String!,
        $level: Int,
        $firstName: String!,
        $middleName: String,
        $lastName: String!,
        $password: String!) {

            createUser(
                username: $username,
                email: $email,
                level: $level,
                firstName: $firstName,
                middleName: $middleName,
                lastName: $lastName,
                password: $password
            ) {
                id
            }
        }
`;

export {
    UserQuery,
    PrivateUserQuery,
    UserUpdate,
    UserLogin,
    RecoverPassword,
    UserCreate,
    UserVerifyEmail
};