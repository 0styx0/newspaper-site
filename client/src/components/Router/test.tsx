import * as React from 'react';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import RouterContainer from './container';
import localStorageMock from '../../tests/localstorage.mock';
import { renderWithGraphql } from '../../tests/snapshot.helper';
import mockGraphql, { createMutation, mountWithGraphql } from '../../tests/graphql.helper'; 
import mainPageSnapData from '../../Pages/MainPage/__snapshots__/issue.example';
import { ArticlePreviewIssueQuery, ArticlePreviewTagQuery } from '../../graphql/articles';
import { IssueInfoQuery } from '../../graphql/issues';
import issueData from '../../Pages/MainPage/__snapshots__/issue.example';
import customCasual from '../../tests/casual.data';
import setFakeJwt from '../../tests/jwt.helper';

describe('<RouterContainer>', () => {

    describe('snapshots', () => {

        async function snap() {

            await renderWithGraphql(
                mockGraphql(
                    [
                        createMutation(ArticlePreviewIssueQuery, { issue: '' }, { issues: [mainPageSnapData] })
                        createMutation(IssueInfoQuery, { num: 0 }, { issues: [issueData] })
                    ],
                    <RouterContainer />
                )
            );
        }

        test('when not logged in, renders only public paths', async () => {

            localStorageMock.clear();
            await snap();
        });

        test('when logged in, renders all paths', async () => {
            setFakeJwt({ level: 3 });
            await snap();
        });
    });

    test(`when not logged in and try to access private path, can't`, async () => {

        localStorageMock.clear();
        const wrapper = await mountWithGraphql(
            [
                createMutation(ArticlePreviewIssueQuery, { issue: '' }, { issues: [mainPageSnapData] })
                createMutation(IssueInfoQuery, { num: 0 }, { issues: [issueData] })
            ],
            <RouterContainer />
            );

        expect(wrapper.find('Link[to="/publish"]')).toHaveLength(0);
    });
});
