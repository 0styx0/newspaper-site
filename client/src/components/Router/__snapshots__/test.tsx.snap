// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouterContainer> snapshots when logged in, renders all paths 1`] = `
<div>
  <nav>
    <ul>
      <label
        htmlFor="menuToggle"
      >
        <span
          className="container"
        />
        <li
          className="showMenu hidden"
        >
           ||| 
        </li>
      </label>
      <input
        id="menuToggle"
        tabIndex={-1}
        type="checkbox"
      />
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <span>
          <select
            defaultValue=""
            onInput={[Function]}
          >
            <option
              value="Current Issue"
            >
              Current Issue
            </option>
          </select>
          
        </span>
      </li>
      
      <li>
        <a
          href="/signup"
          onClick={[Function]}
        >
          Create Account
        </a>
      </li>
      <li>
        <a
          href="/u"
          onClick={[Function]}
        >
          Journalists
        </a>
      </li>
      <li>
        <a
          href="/mission"
          onClick={[Function]}
        >
          Mission
        </a>
      </li>
      <li>
        <a
          href="/issue"
          onClick={[Function]}
        >
          Issues
        </a>
      </li>
      <li>
        <a
          href="/modifyArticles"
          onClick={[Function]}
        >
          Articles
        </a>
      </li>
      <li>
        <a
          href="/publish"
          onClick={[Function]}
        >
          Publish
        </a>
      </li>
      <li
        id="logout"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Logout
          </button>
        </a>
      </li>
      <li
        className="profile"
      >
        <a
          href="/u/undefined"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
    </ul>
  </nav>
  <div>
    (
    ), (
    <div>
      <header>
        <h1>
          <img
            alt="Logo"
            src="/test-logo-url"
          />
          Eye Of The Storm
        </h1>
        <q>
          A Clearer View Of TABC
        </q>
        <h2>
          Aperiam nulla
        </h2>
      </header>
      <div
        id="mainContent"
      >
        <div
          id="slideShow"
        >
          <a
            href="/"
            onAnimationIteration={[Function]}
            onClick={[Function]}
          >
            <img
              alt="Default, logo"
              className="slideshowPic activePic"
              src="/test-logo-url"
            />
          </a>
        </div>
      </div>
      <span
        id="issueRange"
      >
        <a
          href="/issue/1"
          onClick={[Function]}
        >
          1
        </a>
        <a
          href="/issue/2"
          onClick={[Function]}
        >
          2
        </a>
        <a
          href="/issue/3"
          onClick={[Function]}
        >
          3
        </a>
        <a
          href="/issue/4"
          onClick={[Function]}
        >
          4
        </a>
        <a
          href="/issue/5"
          onClick={[Function]}
        >
          5
        </a>
        ...
        <a
          href="/issue/27"
          onClick={[Function]}
        >
          27
        </a>
        <a
          href="/issue/28"
          onClick={[Function]}
        >
          28
        </a>
        <a
          href="/issue/29"
          onClick={[Function]}
        >
          29
        </a>
        <a
          href="/issue/30"
          onClick={[Function]}
        >
          30
        </a>
        <a
          href="/issue/31"
          onClick={[Function]}
        >
          31
        </a>
        <a
          href="/issue/32"
          onClick={[Function]}
        >
          32
        </a>
        ...
        <a
          href="/issue/89"
          onClick={[Function]}
        >
          89
        </a>
        <a
          href="/issue/90"
          onClick={[Function]}
        >
          90
        </a>
        <a
          href="/issue/91"
          onClick={[Function]}
        >
          91
        </a>
      </span>
      <footer
        className="small"
        id="credits"
      >
        Created by 
        <a
          href="https://davidm.me"
        >
          David Meiseles
        </a>
      </footer>
    </div>
  </div>
</div>
`;

exports[`<RouterContainer> snapshots when not logged in, renders only public paths 1`] = `
<div>
  <nav>
    <ul>
      <label
        htmlFor="menuToggle"
      >
        <span
          className="container"
        />
        <li
          className="showMenu hidden"
        >
           ||| 
        </li>
      </label>
      <input
        id="menuToggle"
        tabIndex={-1}
        type="checkbox"
      />
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <span>
          <select
            defaultValue=""
            onInput={[Function]}
          >
            <option
              value="Current Issue"
            >
              Current Issue
            </option>
          </select>
          
        </span>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <li>
        <a
          href="/signup"
          onClick={[Function]}
        >
          Create Account
        </a>
      </li>
      <li>
        <a
          href="/u"
          onClick={[Function]}
        >
          Journalists
        </a>
      </li>
      <li>
        <a
          href="/mission"
          onClick={[Function]}
        >
          Mission
        </a>
      </li>
      <li>
        <a
          href="/issue"
          onClick={[Function]}
        >
          Issues
        </a>
      </li>
      
      
      
      
    </ul>
  </nav>
  <div>
    (
    ), (
    <div>
      <header>
        <h1>
          <img
            alt="Logo"
            src="/test-logo-url"
          />
          Eye Of The Storm
        </h1>
        <q>
          A Clearer View Of TABC
        </q>
        <h2>
          Aperiam nulla
        </h2>
      </header>
      <div
        id="mainContent"
      >
        <div
          id="slideShow"
        >
          <a
            href="/"
            onAnimationIteration={[Function]}
            onClick={[Function]}
          >
            <img
              alt="Default, logo"
              className="slideshowPic activePic"
              src="/test-logo-url"
            />
          </a>
        </div>
      </div>
      <span
        id="issueRange"
      >
        <a
          href="/issue/1"
          onClick={[Function]}
        >
          1
        </a>
        <a
          href="/issue/2"
          onClick={[Function]}
        >
          2
        </a>
        <a
          href="/issue/3"
          onClick={[Function]}
        >
          3
        </a>
        <a
          href="/issue/4"
          onClick={[Function]}
        >
          4
        </a>
        <a
          href="/issue/5"
          onClick={[Function]}
        >
          5
        </a>
        ...
        <a
          href="/issue/27"
          onClick={[Function]}
        >
          27
        </a>
        <a
          href="/issue/28"
          onClick={[Function]}
        >
          28
        </a>
        <a
          href="/issue/29"
          onClick={[Function]}
        >
          29
        </a>
        <a
          href="/issue/30"
          onClick={[Function]}
        >
          30
        </a>
        <a
          href="/issue/31"
          onClick={[Function]}
        >
          31
        </a>
        <a
          href="/issue/32"
          onClick={[Function]}
        >
          32
        </a>
        ...
        <a
          href="/issue/89"
          onClick={[Function]}
        >
          89
        </a>
        <a
          href="/issue/90"
          onClick={[Function]}
        >
          90
        </a>
        <a
          href="/issue/91"
          onClick={[Function]}
        >
          91
        </a>
      </span>
      <footer
        className="small"
        id="credits"
      >
        Created by 
        <a
          href="https://davidm.me"
        >
          David Meiseles
        </a>
      </footer>
    </div>
  </div>
</div>
`;
